### PR TITLE
SerializationString reduce memory usage

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow up #57717.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 2b20c260f48395493178ee3c439a9f070439ba89 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57787/2b20c260f48395493178ee3c439a9f070439ba89/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7183218312">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57787/2b20c260f48395493178ee3c439a9f070439ba89/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/01926_order_by_desc_limit.sql [hunk 1]
```diff
@@ -11,9 +11,11 @@ SETTINGS index_granularity = 1024, index_granularity_bytes = '10Mi';
 INSERT INTO order_by_desc SELECT number, repeat('a', 1024) FROM numbers(1024 * 300);
 OPTIMIZE TABLE order_by_desc FINAL;
 
-SELECT s FROM order_by_desc ORDER BY u DESC LIMIT 10 FORMAT Null;
+SELECT s FROM order_by_desc ORDER BY u DESC LIMIT 10 FORMAT Null
+SETTINGS max_memory_usage = '400M';
```
Comment (alesapin) [inline tests/queries/0_stateless/01926_order_by_desc_limit.sql:15]: Maybe we should disabled this test with santitizers?

Comment (alexey-milovidov) [reply tests/queries/0_stateless/01926_order_by_desc_limit.sql:15]: The calculated memory usage should not depend on sanitizers,
while sanitizers will use more memory internally.

File: src/DataTypes/Serializations/SerializationString.cpp [hunk 2]
```diff
@@ -175,7 +175,7 @@ static NO_INLINE void deserializeBinarySSE2(ColumnString::Chars & data, ColumnSt
         offsets.push_back(offset);
 
         if (unlikely(offset > data.size()))
-            data.resize(roundUpToPowerOfTwoOrZero(std::max(offset, data.size() * 2)));
+            data.resize_exact(roundUpToPowerOfTwoOrZero(std::max(offset, data.size() * 2)));
```
Comment (taiyang-li) [inline src/DataTypes/Serializations/SerializationString.cpp:178]: Do you think it is reasonable to push this rule onto any other similar places where resize was used ?

Comment (kitaisreal) [reply src/DataTypes/Serializations/SerializationString.cpp:178]: If we use PODArray as just buffer and make manual resize in some loop, then yes.

**Pull request merged**